pkg/registry/generic/rest/proxy.go

```diff
@@ -174,23 +174,53 @@ func (h *UpgradeAwareProxyHandler) tryUpgrade(w http.ResponseWriter, req *http.R
 func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 	dialAddr := netutil.CanonicalAddr(h.Location)
 
+	var dialer func(network, addr string) (net.Conn, error)
```

Comment:
Not without subtly changing the behavior in the https case. `tls.Dial` allows tlsConfig to be nil and will treat as the zero config. `tls.Client` requires non-nil tlsConfig, with either `InsecureSkipVerify: true` or `ServerName` provided.